src/pubsub.c

```diff
@@ -660,14 +703,17 @@ NULL
     {
         /* PUBSUB SHARDCHANNELS */
         sds pat = (c->argc == 2) ? NULL : c->argv[2]->ptr;
-        channelList(c,pat,server.pubsubshard_channels);
+        channelList(c,pat,server.pubsubshard_channels,server.cluster_enabled);
     } else if (!strcasecmp(c->argv[1]->ptr,"shardnumsub") && c->argc >= 2) {
         /* PUBSUB SHARDNUMSUB [ShardChannel_1 ... ShardChannel_N] */
         int j;
-
+        unsigned int slot = 0;
         addReplyArrayLen(c, (c->argc-2)*2);
         for (j = 2; j < c->argc; j++) {
-            list *l = dictFetchValue(server.pubsubshard_channels, c->argv[j]);
+            if (server.cluster_enabled) {
+                slot = keyHashSlot(c->argv[j]->ptr, sdslen(c->argv[j]->ptr));
```

Comment:
Same as above `keyHashSlot` compute comment.

Reply:
Just checked redis.io

> Cluster note: in a Redis Cluster, [PUBSUB](https://redis.io/commands/pubsub)'s replies in a cluster only report information from the node's Pub/Sub context, rather than the entire cluster.

So I guess it is not a bug, and therefore c->slot will not have a computed slot here.

Reply:
For non cluster mode, it would be set to -1.